ibc/store/provable_store.go

```diff
@@ -0,0 +1,244 @@
+	return createMembershipProof(lazy, prefixed, value)
+	**/
+	return nil, nil
+}
+
+// CreateNonMembershipProof creates a non-membership proof for the key prefixed with the
+// CommitmentPrefix, by importing the state tree from the TreeStore
+func (p *provableStore) CreateNonMembershipProof(key []byte) (*ics23.CommitmentProof, error) {
+	// import IBC state tree
+	// TODO(#854): Implement tree retrieval
+	/**
+	prefixed := applyPrefix(p.prefix, key)
+	root, nodeStore := p.bus.GetTreeStore().GetTree(trees.IBCStateTree)
+	lazy := smt.ImportSparseMerkleTree(nodeStore, root, sha256.New())
+	return createNonMembershipProof(lazy, prefixed)
+	**/
+	return nil, nil
+}
+
+// Set updates the persistence layer with the new key-value pair at the latest height and
+// emits an UpdateIBCStore event to the bus for propagation throughout the network, to be
+// included in each node's mempool and thus the next block
+func (p *provableStore) Set(key, value []byte) error {
+	prefixed := applyPrefix(p.prefix, key)
+	currHeight := int64(p.bus.GetConsensusModule().CurrentHeight())
+	rwCtx, err := p.bus.GetPersistenceModule().NewRWContext(currHeight)
+	if err != nil {
+		return err
+	}
+	defer rwCtx.Release()
+	if err := rwCtx.SetIBCStoreEntry(prefixed, value); err != nil {
+		return err
+	}
+	p.cache = append(p.cache, &cachedEntry{
+		storeName:   p.name,
+		height:      uint64(currHeight),
+		prefixedKey: prefixed,
+		value:       value,
+	})
+	// TODO(#854): Implement emit functions
+	// return emitUpdateStoreEvent(p.prefix, key, value)
+	return nil
+}
+
+// Delete updates the persistence layer with the key and nil value pair at the latest height
+// and emits an PruneIBCStore event to the bus for propagation throughout the network, to be
+// included in each node's mempool and thus the next block
+func (p *provableStore) Delete(key []byte) error {
+	prefixed := applyPrefix(p.prefix, key)
+	currHeight := int64(p.bus.GetConsensusModule().CurrentHeight())
+	rwCtx, err := p.bus.GetPersistenceModule().NewRWContext(currHeight)
+	if err != nil {
+		return err
+	}
+	defer rwCtx.Release()
+	if err := rwCtx.SetIBCStoreEntry(prefixed, nil); err != nil {
+		return err
+	}
+	p.cache = append(p.cache, &cachedEntry{
+		storeName:   p.name,
+		height:      uint64(currHeight),
+		prefixedKey: prefixed,
+		value:       nil,
+	})
+	// TODO(#854): Implement emit functions
+	// return emitPruneStoreEvent(p.prefix, key)
+	return nil
+}
+
+// GetCommitmentPrefix returns the CommitmentPrefix for the store
+func (p *provableStore) GetCommitmentPrefix() coreTypes.CommitmentPrefix { return p.prefix }
+
+// Root returns the current root of the IBC state tree
+func (p *provableStore) Root() ics23.CommitmentRoot {
+	// TODO(#854): Implement tree retrieval
+	/**
+	root, _ := p.bus.GetTreeStore().GetTree(trees.IBCStateTree)
+	return root
+	**/
+	return nil
+}
+
+// CacheEntries writes all local changes to disk and clears the in-memory cache
```

Comment:
s/CacheEntries/FlushEntries

Flush is the proper term for this since you're moving from cache to disk
https://stackoverflow.com/a/48331611/768439